apps/vaporgui/PMetadataSection.cpp

```diff
@@ -0,0 +1,145 @@
+#include <QTreeWidget>
+#include <QHeaderView>
+#include <sstream>
+
+#include "ErrorReporter.h"
+#include "PMetadataSection.h"
+#include "PSection.h"
+
+#include <vapor/RenderParams.h>
+#include <vapor/DataMgr.h>
+#include <vapor/DC.h>
+
+// clang-format off
+
+namespace {
+    std::vector<std::string> xtypeLookup = { "INVALID", "FLOAT", "DOUBLE", "UINT8", "INT8", "INT32", "INT64", "TEXT" };
+    std::vector<std::string> axisLookup  = { "X", "Y", "Z", "T"};
+}
+
+PMetadataSection::PMetadataSection()
+: PSection("Metadata", { new PMetadataTree() } )
+{
+}
+
+PMetadataTree::PMetadataTree()
+: PWidget("", _tree = new QTreeWidget)
+{
+    _tree->setHeaderHidden(true);
+    _tree->header()->setStretchLastSection(false);
+    _tree->setColumnCount(2);
+    _tree->header()->setSectionResizeMode(QHeaderView::ResizeToContents);
+}
+
+void PMetadataTree::updateGUI() const
+{
+    VAPoR::RenderParams* rp = dynamic_cast<VAPoR::RenderParams*>(getParams());
+    if (rp == nullptr) MSG_ERR("Invalid RenderParams instance for acquiring variable metadata.");
+    VAPoR::DataMgr* dm = getDataMgr();
+    if (rp == nullptr) MSG_ERR("Invalid DataMgr instance for acquiring variable metadata.");
+
+    std::vector<std::string> vars;
+    vars.push_back( rp->GetVariableName() );
+    vars.push_back( rp->GetXFieldVariableName() );
+    vars.push_back( rp->GetYFieldVariableName() );
+    vars.push_back( rp->GetZFieldVariableName() );
+    vars.push_back( rp->GetHeightVariableName() );
+    vars.push_back( rp->GetColorMapVariableName() );
+    
+    std::vector<std::string> auxs = rp->GetAuxVariableNames();
+    vars.insert(vars.end(), auxs.begin(), auxs.end());
+
+    QStringList qvars;
+    for (auto var : vars)
+        qvars << QString::fromStdString(var);
+    qvars.removeDuplicates();
+    qvars.removeAll(QString(""));
+
+    size_t ts = rp->GetCurrentTimestep();
+    int rLevel = rp->GetRefinementLevel();
+    int cLevel = rp->GetCompressionLevel();
+
+    _tree->clear();
+    for (auto qvar : qvars) {
+        std::vector<double> range;
+        dm->GetDataRange(ts, qvar.toStdString(), rLevel, cLevel, range);
```

Comment:
GetDataRange() is an expensive operation. Can this be done lazily? I.e. only gather metadata for variables that are "open" in the display window rather than calling GetDataRange() on potentially multiple variables that the user may or may not be looking at?

Reply:
I'm finding problems with this.  The QTreeWidget is cleared and regenerated on each call to PMetadataTree::updateGUI(), and the branches do not persist across params updates.

Ideally we would only reintegrate the tree when there is a change to 
1) the current time step
2) a variable's rLevel or cLevel
3) a new variable is selected.  

But because updateGUI() is const, we cannot update/refresh member variables that would track these values.

@clyne and @stasj, what do you think about the following:

Add a function to the PWidget base class, **_needsUpdate()**, that can be overridden, and includes logic that can prevent calls to a PWidget's updateGUI.  The default implementation would return true.  This would then be called at [PWidget.cpp::42](https://github.com/NCAR/VAPOR/blob/f80b3e8af6007203b3e8bd82019a6384d319dfd9/apps/vaporgui/PWidget.cpp#L42).

Reply:
The dataset metadata display could be a V-Widget

Reply:
I will defer to @stasj on this.